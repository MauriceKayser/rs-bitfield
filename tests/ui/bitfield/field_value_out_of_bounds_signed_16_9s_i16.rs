//! Assertion failure: _TYPE_IN_FIELD_0_EXCEEDS_FIELD_SIZE_OF_9_BITS

extern crate alloc;

#[bitfield::bitfield(16)]
struct BitField(#[field(size = 9, signed)] Field);

#[derive(Clone, Copy, Debug, bitfield::Field)]
#[repr(i16)]
enum Field {
    F2 = -300
}

fn main() {}