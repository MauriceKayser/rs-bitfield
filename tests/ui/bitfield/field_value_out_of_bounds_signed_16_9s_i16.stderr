error[E0080]: evaluation of constant value failed
 --> tests/ui/bitfield/field_value_out_of_bounds_signed_16_9s_i16.rs:6:44
  |
6 | struct BitField(#[field(size = 9, signed)] Field);
  |                                            ^^^^^ the evaluated program panicked at 'Size of the field type exceeds the specified field size of 9 bits', $DIR/tests/ui/bitfield/field_value_out_of_bounds_signed_16_9s_i16.rs:6:44
  |
  = note: this error originates in the macro `$crate::panic::panic_2021` which comes from the expansion of the macro `panic` (in Nightly builds, run with -Z macro-backtrace for more info)
